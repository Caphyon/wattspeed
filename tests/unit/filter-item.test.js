
import { shallowMount } from "@vue/test-utils";
import filter_item from "../../src/components/panels/tech/items/filter-item.vue"; // name of your Vue component

describe("filter-item.vue", () => {

  it("Testing filter-item component", () => {
    const wrapper = shallowMount(filter_item, {
      propsData:{
        filterCategories: ['Success', 'Warnings', 'Errors']
      },
    });
    wrapper.setData({
      Success: 'Success',
      Warnings: 'Warnings',
      Errors: 'Errors'
    })

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.findAll('label').length).toBe(3);
    expect(wrapper.find('.cursor--pointer.filter__checkbox.filter--success').exists()).toBe(true);
    expect(wrapper.find('.cursor--pointer.filter__checkbox.filter--warning').exists()).toBe(true);
    expect(wrapper.find('.cursor--pointer.filter__checkbox.filter--danger').exists()).toBe(true);

  })
});
