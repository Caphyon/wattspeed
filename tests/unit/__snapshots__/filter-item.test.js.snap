// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter-item.vue Testing filter-item component 1`] = `
<div
  class="text--right text--extrasmall py1"
>
  <label
    class="cursor--pointer filter__checkbox filter--success text--success"
    title="Toggle success checks"
  />
   
  <label
    class="cursor--pointer filter__checkbox filter--warning text--warning"
    title="Toggle warning checks"
  />
   
  <label
    class="cursor--pointer filter__checkbox filter--danger text--danger"
    title="Toggle error checks"
  />
</div>
`;
